src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.cs

```diff
@@ -501,9 +517,38 @@ private bool IsFirstTokenOnLine(TokenData tokenData1, TokenData tokenData2)
             return this.GetTriviaData(tokenData1, tokenData2).SecondTokenIsFirstTokenOnLine;
         }
 
+        /// <summary>
+        /// Gets approximated index bounds for a token with specified position, which will definitely contain
+        /// the token, if it is present in the stream.
+        /// </summary>
+        private void GetTokenIndexBounds(int tokenFullSpanStart, out int lowerBound, out int upperBound)
+        {
+            Debug.Assert(_averageTokenLength > 0);
+            // make an initial guess about the token position
+            lowerBound = upperBound = Bounded((int)((tokenFullSpanStart - _tokens[0].FullSpan.Start) / _averageTokenLength), min: 0, max: _tokens.Length - 1);
+
+            // decrease lower bound until the desired position is strictly after it
+            while (_tokens[lowerBound].FullSpan.Start >= tokenFullSpanStart && lowerBound > 0)
```

Comment:
i'd add a comment on this why `>=` is necessary over `>`.  Specifically, `>` would likely not be sufficient, because we'd stop when hitting a token that starts at `tokenFullSpanStart`.  However (due to missing tokens) there may be several tokens that start at that location, so we need to keep walking backward in order to capture them all.  But doing `>=` we'll walk to hit the first token that precedes *all* the potential tokens at that position.